autojustice.lua

```diff
@@ -0,0 +1,297 @@
+-- Auto Justice
+--@enable = true
+
+local usage = [[
+Usage
+-----
+enable autojustice
+disable autojustice
+]]
+
+local GLOBAL_KEY = 'autojustice'
+local UNIT_NEW_ACTIVE_DELAY = 100
+local CRIME_REFRESH_DELAY = 120
+
+local json = require('json')
+local persist = require('persist-table')
+local repeatUtil = require('repeat-util')
+local eventful = require('plugins.eventful')
+local justice = reqscript('autojustice/justicetools')
+
+enabled = enabled or false
+
+local lastCrimeCount = 0
+local openCrimes = nil
+local openCrimesWitnessCount = nil
+local undiscoveredCrimes = nil
+local confessedCrimes = nil
+
+function isEnabled()
+    return enabled
+end
+
+function saveState()
+    persist.GlobalTable[GLOBAL_KEY] = json.encode({
+        enabled = enabled or false
+    })
+end
+
+function loadState ()
+    local data = json.decode(persist.GlobalTable[GLOBAL_KEY] or '{}') 
+    enabled = data.enabled or false
+end
+
+function onStateChange (sc)
```

Comment:
this should be hooked up to dfhack directly -- `dfhack.onStateChange[GLOBAL_KEY] = function(sc)`
otherwise your reload code won't actually restore the enabled state for loaded forts automatically. You can copy the structure from `emigration.lua` for this.

Reply:
note that the eventful config should still be where it is in registerEvents() since that should only be run when autojustice is enabled